src/document_change_manager.ts

```diff
@@ -95,7 +95,13 @@ export class DocumentChangeManager implements Disposable, NeovimExtensionRequest
     public constructor(private main: MainController) {
         this.main.bufferManager.onBufferEvent = this.onNeovimChangeEvent;
         this.main.bufferManager.onBufferInit = this.onBufferInit;
-        this.disposables.push(workspace.onDidChangeTextDocument(this.onChangeTextDocument));
+        this.disposables.push(
+            workspace.onDidChangeTextDocument(this.onChangeTextDocument),
+            eventBus.on("insert-line", ([type]) => {
```

Comment:
`insert-line` is actually no longer used so we can delete it here and elsewhere. However, this might be better to have in a different document-change-manager cleanup PR (I've already cleaned it up in a few different PRs I was unable to merge :upside_down_face: )

Reply:
I know this. It's very rough at the moment and needs to be refactored in several PRs.